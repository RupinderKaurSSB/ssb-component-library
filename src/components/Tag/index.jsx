import React from 'react';
import PropTypes from 'prop-types';

const Tag = ({
	children, className, icon, onClick,
}) => (
	<span
		className={`ssb-tag${className ? ` ${className}` : ''}`}
		onClick={onClick}
		role="button"
		tabIndex={0}
	>
		{icon && <div className="st-icon">{icon}</div>}
		{children}
	</span>
);

Tag.defaultProps = {
	className: '',
	onClick: () => {},
};

Tag.propTypes = {
	children: PropTypes.node.isRequired,
	className: PropTypes.string,
	icon: PropTypes.node,
	onClick: PropTypes.func,
};

export default Tag;
