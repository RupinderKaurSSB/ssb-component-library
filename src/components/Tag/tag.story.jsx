import React from 'react';
import { storiesOf } from '@storybook/react';
import centered from '@storybook/addon-centered/react';
import { Star } from 'react-feather';
import Tag from './index';

storiesOf('Tag', module).addDecorator(centered)
	.add('Tag', () => (
		<div style={{ display: 'flex' }}>
			<Tag>Tag</Tag>
		</div>
	))
	.add('Tag with icon', () => (
		<div style={{ display: 'flex' }}>
			<Tag icon={<Star size={18} />}>Tag with icon</Tag>
		</div>
	));
