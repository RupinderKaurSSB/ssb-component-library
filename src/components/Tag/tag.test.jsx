import React from 'react';
import {shallow} from 'enzyme';
import Tag from './index';

describe('Tag component', () => {
	test('Matches the snapshot', () => {
		const wrapper = shallow(<Tag>Tag</Tag>);
		expect(wrapper).toMatchSnapshot();
	});
	test('Renders icon', () => {
		const wrapper = shallow(<Tag icon={<i />}>Tag</Tag>);
		expect(wrapper.find('.ssb-tag').containsMatchingElement(<i />)).toEqual(true);
	});
	test('Sends callback', () => {
		const onClick = jest.fn();
		const wrapper = shallow(<Tag onClick={onClick}>Tests</Tag>);
		wrapper.find('.ssb-tag').simulate('click');
		expect(onClick).toBeCalled();
		wrapper.find('.ssb-tag').simulate('click');
		expect(onClick).toBeCalledTimes(2);
	});
});
