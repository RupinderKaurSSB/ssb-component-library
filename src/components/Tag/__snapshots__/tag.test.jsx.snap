// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag component Matches the snapshot 1`] = `
<span
  className="ssb-tag "
  onClick={[Function]}
>
  Tag
</span>
`;
